qiskit/circuit/quantumcircuit.py

```diff
@@ -285,6 +286,45 @@ def __init__(
             raise TypeError("Only a dictionary or None is accepted for circuit metadata")
         self._metadata = metadata
 
+    @staticmethod
+    def from_instructions(
+        instructions: Iterable[
+            CircuitInstruction | tuple[Instruction, Iterable[Qubit], Iterable[Clbit]]
+        ],
```

Comment:
By analogy to the signatures of `append` and the constructor of `CircuitInstruction`, do you think it would be worthwhile to accept the case where the clbits (and _in theory_) the qubits can be omitted from the tuple?  I'm ok either way.  I'm _marginally_ in favour of allowing the last two items in the tuple to be optional, but not very strongly.  If we did do that, I meant a signature like `instruction, [qubits, [clbits]]` (using man-page notation), i.e. you can drop the clbits, or the qubits _and_ clbits, but never just the qubits.  I don't know if Python's type hinting language can represent that well, though - we'd probably get stuck with an unholy
```
CircuitInstruction | tuple[Instruction] | tuple[Instruction, Iterable[Qubit]] | tuple[Instruction, Iterable[Qubit], Iterable[Clbit]]
```
or just have to say
```
CircuitInstruction | Iterable[tuple]
```
and explain in the comments.

Reply:
Done.